segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/ReadOperation.java

```diff
@@ -120,6 +121,7 @@ private void logEnd() {
     }
 
     private CompletableFuture<Void> readData(MetadataTransaction txn) {
+        val chunkReadFutures = new ArrayList<CompletableFuture<Void>>();
         return Futures.loop(
                 () -> bytesRemaining.get() > 0 && null != currentChunkName,
```

Comment:
Is there an upper bound on the number of parallel reads? We should always put a ceiling on such requests to prevent some corner case to bring down the host application or the remote server.

Reply:
Here the number of requests is bound by number of thread pool size. Whether there are 1 chunk reads on 200 segments, or 200 chunk reads on 1 segment, the number of outstanding I/O requests can not exceed thread pool size.